omod-common/src/test/java/org/openmrs/module/webservices/rest/web/filter/ContentTypeFilterTest.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.filter;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.mock.web.MockFilterChain;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Tests for the {@link ContentTypeFilter} class.
+ */
+public class ContentTypeFilterTest {
+	
+	private ContentTypeFilter testFilter;
+	
+	private MockFilterChain mockChain;
+	
+	private MockHttpServletRequest req;
+	
+	private MockHttpServletResponse resp;
+	
+	@Before
+	public void init() {
+		testFilter = new ContentTypeFilter();
+		mockChain = new MockFilterChain();
+		req = new MockHttpServletRequest();
+		resp = new MockHttpServletResponse();
+	}
+	
+	@Test
+	public void doFilter_shouldNotAllowNonJsonContent() throws IOException, ServletException {
+		
+		List<String> xmlContentTypes = Arrays.asList("application/xml", "text/xml", "application/zip", "fake/contenttype");
+		
+		for (String contentType : xmlContentTypes) {
+			init();
+			req.setContentType(contentType);
+			req.setMethod("POST");
+			req.setRequestURI("/ws/rest/v1/obs");
+			testFilter.doFilter(req, resp, mockChain);
+			
+			Assert.assertEquals(resp.getStatus(), HttpServletResponse.SC_UNSUPPORTED_MEDIA_TYPE);
+		}
+	}
+	
+	@Test
+	public void doFilter_shouldAllowJSONContent() throws IOException, ServletException {
+		req.setContentType("application/json");
+		req.setMethod("POST");
+		req.setRequestURI("/ws/rest/v1/obs");
+		testFilter.doFilter(req, resp, mockChain);
+		
+		Assert.assertNotEquals(resp.getStatus(), HttpServletResponse.SC_UNSUPPORTED_MEDIA_TYPE);
+	}
+	
+	@Test
+	public void doFilter_shouldAllowGetRequest() throws IOException, ServletException {
```

Comment:
As for the methods name, should allow Get request for all content types?

Reply:
According to the RFCs, a GET request shouldn't have a Content-Type, but I think an attacker could theoretically craft a GET request with a Content-Type header that may trigger this vulnerability. So I think two things are important here:

1. We make sure we're allowing RFC-compliant GET requests because they are crucial to the normal operation of the Refapp.
2. We make no assumptions about RFC-compliance when security checking and only allow requests with json or null Content-Types, regardless of the type of request (GET, POST, PUT, etc.)

With that in mind, would it be more appropriate to split that into two tests? E.g. doFilter_shouldAllowGetRequest and doFilter_shouldAllowNullContentType?

Reply:
makes sense